plugin-api/src/main/java/org/hyperledger/besu/plugin/data/TransactionReceipt.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.plugin.data;
+
+import java.util.Optional;
+
+import org.apache.tuweni.bytes.Bytes;
+
+/** A transaction receipt, containing information pertaining a transaction execution. */
+public interface TransactionReceipt {
+  Optional<Bytes> getRevertReason();
```

Comment:
I think we should add the 4 fields listed in section 4.3.1 of the yellow paper:   cumulativeGasUsed, logs, bloomFilter, and status.  For bloomFilter I think we want to just expose the Bytes object rather than our LogBloomFilter.  If we get covariant type complaints from Java we can just create a new method (BloomFilterBytes) and cast it down., but for the other three we can just expose the existing methods, since we have Log already as a data type.

Also, they all the methods need javadoc.  RevertReason is our own field so we should mention it's data beyond what the yellow paper requires, but for the other four just stick with the simple definitions.